wp-content/plugins/tapestry/controller/class.user-controller.php

```diff
@@ -0,0 +1,143 @@
+<?php
+// TODO Change exceptions to using an ERROR class
+/**
+ * Add/update/retrieve User progress
+ * 
+ */
+class TapestryUserController {
+
+    private $userId = NULL;
+
+    public function __construct() {
+        $this->userId = apply_filters('determine_current_user', false); 
+    }
+
+    /** 
+     * Update User's video progress for a tapestry post
+     * @param type @postId the post's ID
+     * @param type @nodeId the current node being viewed
+     * @param type @progressValue is how much the video was viewed, value should be between >= 0 and <= 1
+    */
+    public function updateProgress($postId, $nodeId, $progressValue) {
+        $this->_checkUserAndPostId($this->userId, $postId);
+
+        if ($progressValue !== NULL) {
+            $progressValue = json_decode($progressValue);
+        }
+
+        // Value should be between 0 and 1
+        if ($progressValue < 0 || $progressValue > 1) {
+            throw new Exception('Invalid progress value');
+        }
+
+        $this->_updateUserProgress($this->userId, $postId, $nodeId, $progressValue);
```

Comment:
You shouldn't be passing class properties to your methods since they are already accessible from your methods. Please fix this everywhere.

Reply:
Removing passing userId